pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -35,14 +35,27 @@ patchShebangs() {
                 echo "unsupported interpreter directive \"$oldInterpreterLine\" (set dontPatchShebangs=1 and handle shebang patching yourself)"
                 exit 1
             fi
-            newPath="$(command -v "$arg0" || true)"
+
+            (
+                if [ -n "$strictDeps" ] && [[ "$f" = /nix/store/* ]]; then
+                    PATH="$HOST_PATH"
+                fi
+                newPath="$(command -v "$arg0" || true)"
+            )
         else
             if [ "$oldPath" = "" ]; then
                 # If no interpreter is specified linux will use /bin/sh. Set
                 # oldpath="/bin/sh" so that we get /nix/store/.../sh.
                 oldPath="/bin/sh"
             fi
-            newPath="$(command -v "$(basename "$oldPath")" || true)"
+
+            (
+                if [ -n "$strictDeps" ] && [[ "$f" = /nix/store/* ]]; then
+                    PATH="$HOST_PATH"
+                fi
+                newPath="$(command -v "$(basename "$oldPath")" || true)"
```

Comment:
Just realized that basename is not a bash builtin :(. Have to rewrite it to run outside of subshell.

Reply:
You can't also hardcode `/nix/store`.

Reply:
The hardcoding may be necessary for back compat - but I agree that we will probably need a way to be specific about what we are targeting with patchShebangs. Something like an optional offset argument.

Reply:
I mean you need to use `$NIX_STORE_DIR` or something.